src/main/java/emu/grasscutter/server/packet/recv/HandlerCombatInvocationsNotify.java

```diff
@@ -84,33 +85,42 @@ public void handle(GameSession session, byte[] header, byte[] payload) throws Ex
 	}
 
 	private void handleFallOnGround(GameSession session, GameEntity entity, MotionState motionState) {
-		// If not received immediately after MOTION_LAND_SPEED, discard this packet.
+		// People have reported that after plunge attack (client sends a FIGHT instead of FALL_ON_GROUND) they will die
+		// 		if they talk to an NPC (this is when the client sends a FALL_ON_GROUND) without jumping again.
+		// A dirty patch: if not received immediately after MOTION_LAND_SPEED, discard this packet.
+		// 200ms seems to be a reasonable delay.
 		int maxDelay = 200;
 		long actualDelay = System.currentTimeMillis() - cachedLandingTimeMillisecond;
-		Grasscutter.getLogger().debug("MOTION_FALL_ON_GROUND received after " + actualDelay + "/" + maxDelay + "ms." + (actualDelay > maxDelay ? " Discard" : ""));
+		Grasscutter.getLogger().trace("MOTION_FALL_ON_GROUND received after " + actualDelay + "/" + maxDelay + "ms." + (actualDelay > maxDelay ? " Discard" : ""));
 		if (actualDelay > maxDelay) {
 			return;
 		}
 		float currentHP = entity.getFightProperty(FightProperty.FIGHT_PROP_CUR_HP);
 		float maxHP = entity.getFightProperty(FightProperty.FIGHT_PROP_MAX_HP);
-		float damage = 0;
+		float damageFactor = 0;
 		if (cachedLandingSpeed < -23.5) {
-			damage = (float) (maxHP * 0.33);
+			damageFactor = 0.33f;
 		}
 		if (cachedLandingSpeed < -25) {
-			damage = (float) (maxHP * 0.5);
+			damageFactor = 0.5f;
 		}
 		if (cachedLandingSpeed < -26.5) {
-			damage = (float) (maxHP * 0.66);
+			damageFactor = 0.66f;
 		}
 		if (cachedLandingSpeed < -28) {
-			damage = (maxHP * 1);
+			damageFactor = 1f;
 		}
+		float damage = maxHP * damageFactor;
 		float newHP = currentHP - damage;
 		if (newHP < 0) {
 			newHP = 0;
 		}
-		Grasscutter.getLogger().debug(currentHP + "/" + maxHP + "\t" + "\tDamage: " + damage + "\tnewHP: " + newHP);
+		if (damageFactor > 0) {
+			Grasscutter.getLogger().debug(currentHP + "/" + maxHP + "\tLandingSpeed: " + cachedLandingSpeed +
```

Comment:
Don't see particular reason to keep `DEBUG` here specifically for `damageFactor > 0`, however regardless of whether it'll be kept or not, there should be some indication on where the log message is coming from, as it is otherwise not clear at all (just bunch of numbers).